### PR TITLE
Add a callback to skip the opening of a recursive directory

### PR DESCRIPTION
**Pull request by Amatewasu**
This PR allows passing a callback to say whether we want to open this directory based on the developer criteria (such as the name for example). For example, you can skip the directory with a name beginning with a point ('.git', '.backup', etc.).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,6 +120,8 @@ const options = {
   startIn: 'downloads',
   // By specifying an ID, the user agent can remember different directories for different IDs.
   id: 'projects',
+  // Callback whether a directory needs to be opened, return true to skip
```
Comment (tomayac) [inline README.md:123]:
```suggestion
  // Callback to determine whether a directory should be entered, return `true` to skip.
```

File: index.d.ts [hunk 2]
```diff
@@ -161,6 +161,8 @@ export function directoryOpen(options?: {
   startIn?: WellKnownDirectory | FileSystemHandle;
   /** By specifying an ID, the user agent can remember different directories for different IDs. */
   id?: string;
+  /** Callback whether a directory needs to be opened, return false to skip */
```
Comment (tomayac) [inline index.d.ts:164]:
```suggestion
  /** Callback to determine whether a directory should be entered, return `true` to skip. */
```

File: src/legacy/directory-open.mjs [hunk 3]
```diff
@@ -49,7 +49,14 @@ export default async (options = [{}]) => {
         files = files.filter((file) => {
           return file.webkitRelativePath.split('/').length === 2;
         });
+      } else if (options[0].recursive && options[0].skipDirectory) {
+        files = files.filter((file) => {
+          const directoriesName = file.webkitRelativePath.split('/');
+
```
Comment (tomayac) [inline src/legacy/directory-open.mjs:55]:
```suggestion
```

**Comment by tomayac** [comment]
Getting there, see my remarks. Can I also ask you to run `npm run fix` and commit the result, please?

**Comment by Amatewasu** [comment]
Sure thank you for your remarks, I committed the result.

**Comment by tomayac** [comment]
(Sorry for the delay, I'm busy with Chrome Dev Summit and other things, but I will get back to you, promised.)

**Comment by tomayac** [comment]
Thank you again! This has now been published as `v0.21.1`.

**Pull request merged**